includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,489 @@
+		$story_data['poster']               = get_the_post_thumbnail_url( $story_id, $image_size );
+		$story_data['author']               = $author_name;
+		$story_data['date']                 = $story_date;
+		$story_data['class']                = $this->get_single_story_classes();
+		$story_data['show_content_overlay'] = ( ! empty( $story_title ) || ! empty( $author_name ) || ! empty( $story_date ) );
+
+		return $story_data;
+	}
+
+	/**
+	 * Verifies the current view type.
+	 *
+	 * @param string $view_type View type to check.
+	 *
+	 * @return bool Whether or not current view type matches the one passed.
+	 */
+	protected function is_view_type( $view_type ) {
+
+		return ( ! empty( $this->attributes['view_type'] ) && $view_type === $this->attributes['view_type'] );
+	}
+
+	/**
+	 * Get view type for stories.
+	 *
+	 * @return string
+	 */
+	protected function get_view_type() {
+
+		return ( ! empty( $this->attributes['view_type'] ) ) ? $this->attributes['view_type'] : 'circles';
+	}
+
+	/**
+	 * Renders stories archive link if the 'show_stories_archive_link' attribute is set to true.
+	 *
+	 * @return void
+	 */
+	protected function maybe_render_archive_link() {
+
+		if ( empty( $this->attributes['show_stories_archive_link'] ) || true !== $this->attributes['show_stories_archive_link'] ) {
+			return;
+		}
+
+		$web_stories_archive = get_post_type_archive_link( Story_Post_Type::POST_TYPE_SLUG );
+
+		if ( empty( $web_stories_archive ) || ! is_string( $web_stories_archive ) ) {
+			return;
+		}
+
+		?>
+		<div class="web-stories-list__archive-link">
+			<a href="<?php echo esc_url( $web_stories_archive ); ?>">
+				<?php echo esc_html( $this->attributes['stories_archive_label'] ); ?>
+			</a>
+		</div>
+		<?php
+
+	}
+
+	/**
+	 * Gets the classes for renderer container.
+	 *
+	 * @return string
+	 */
+	protected function get_container_classes() {
+
+		$container_classes   = [];
+		$container_classes[] = 'web-stories-list';
+		$container_classes[] = ( ! empty( $this->attributes['view_type'] ) ) ? sprintf( 'is-view-type-%1$s', $this->attributes['view_type'] ) : 'is-view-type-circles';
+		$container_classes[] = ( ! empty( $this->attributes['align'] ) ) ? sprintf( 'align%1$s', $this->attributes['align'] ) : 'alignnone';
+		$container_classes[] = ( ! empty( $this->attributes['class'] ) ) ? $this->attributes['class'] : '';
+
+		$container_classes = array_filter( $container_classes );
+
+		return implode( ' ', $container_classes );
+	}
+
+	/**
+	 * Gets the single story container classes.
+	 *
+	 * @return string
+	 */
+	protected function get_single_story_classes() {
+
+		$single_story_classes   = [];
+		$single_story_classes[] = 'web-stories-list__story-wrapper';
+
+		if ( ! $this->is_view_type( 'grid' ) ) {
+			$single_story_classes[] = 'has-poster';
+		}
+
+		if ( ! empty( $this->attributes['show_story_poster'] ) &&
+			$this->is_view_type( 'grid' ) &&
+			true === $this->attributes['show_story_poster']
+		) {
```

Comment:
Seems like it is repeated logic isnt `true === $this->attributes['show_story_poster']` enough?